config/apcupsd/apcupsd.inc

```diff
@@ -115,6 +115,28 @@ function validate_input_apcupsd($post,&$input_errors){
 			}	
 		}
 
+		if ($post['emailnotification'] == 'smtpbuiltin') {
+
+			if (empty($post['smtphost'])) {
+					$input_errors[]='SMTP Host cannot be empty if SMTP builtin is enabled.';
+			}
+
+			if (empty($post['smtpport'])) {
+					$input_errors[]='SMTP Port cannot be empty if SMTP builtin is enabled.';
+			}elseif(!is_numericint($post['smtpport'])){
```

Comment:
There is a function called is_port() at util.inc, you should use it to validate port